### PR TITLE
Added the close button

### PR DESCRIPTION
**Pull request by PawBud**
This is a fix for the issue: _#1579_

@4ian, I am sure that I have added the close button physically but it's not working. I just wanted some help with my code. Can you tell me in brief as to what components do I have to add? Moreover, is the approach that I am taking correct in the first place or is there a completely different 
workaround?
I added the `Flatbutton` element in `EventsBasedBehaviorEditorDialog.js` And added the necessary changes to the files where `onClose` was a prop.
Thanks :)

### CODE REVIEW

File: newIDE/app/src/EventsFunctionsExtensionEditor/index.js [hunk 1]
```diff
@@ -862,6 +862,7 @@ export default class EventsFunctionsExtensionEditor extends React.Component<
             project={project}
             eventsFunctionsExtension={eventsFunctionsExtension}
             eventsBasedBehavior={editedEventsBasedBehavior}
+            onClose={() => this._editBehavior(null)}
```
Comment (PawBud) [inline newIDE/app/src/EventsFunctionsExtensionEditor/index.js]: I am sure this needs to change

**Comment by PawBud** [comment]
@Bouh @4ian May I get some help?

**Comment by 4ian** [comment]
This is fairly complex, that's why I've not marked #1579 as a Good First Issue. You probably want to use:
* `withSerializableObject` to make a copy of the behavior. It will handle the apply operation for you. 
* You need to do the job of `onRenameProperty` for all the properties that were renamed in the apply function (i.e: when the user closes by clicking on Apply, then `withSerializableObject` will update the behavior and you also need to get the list of renamed properties and call this._onBehaviorPropertyRenamed

**Comment by PawBud** [comment]
thank a lot, I will look into it :).

**Comment by PawBud** [comment]
> This is fairly complex, that's why I've not marked #1579 as a Good First Issue. You probably want to use:
> 
> * `withSerializableObject` to make a copy of the behavior. It will handle the apply operation for you. Check in the codebase how it's used.
> * You need to do the job of `onRenameProperty` for all the properties that were renamed in the apply function (i.e: when the user closes by clicking on Apply, then `withSerializableObject` will update the behavior and you also need to get the list of renamed properties and call this._onBehaviorPropertyRenamed

**Pull request closed without merging**